rules/require-signal-reference-ngb-modal-input.mjs

```diff
@@ -0,0 +1,202 @@
+            );
+            // The component symbol might be marked as an alias when it was imported using a different name.
+            // flags is a bit flag enum, so the check is done using bitwise AND.
+            if (symbol?.flags & ts.SymbolFlags.Alias) {
+                symbol = typeChecker.getAliasedSymbol(symbol);
+            }
+            return symbol;
+        };
+
+
+        const extractModalReferenceAssignedVariableName = parentNode => {
+            if (
+                parentNode?.type === "VariableDeclarator" &&
+                parentNode.id.type === "Identifier"
+            ) {
+                return parentNode.id.name;
+            }
+            if (
+                parentNode?.type === "AssignmentExpression" &&
+                parentNode.left.type === "Identifier"
+            ) {
+                return parentNode.left.name;
+            }
+            return null;
+        };
+
+        /**
+         * STEP 1: Capture modal open calls
+         */
+        const handleOpenCallExpression = node => {
+            if (!isModalServiceOpenCall(node)) return;
+            const callee = node.callee;
+            if (!isNgbModalInstance(callee.object)) return;
+
+            const componentSymbol = resolveComponentSymbol(node.arguments[0]);
+            if (!componentSymbol) return;
+
+            const modalVariableName = extractModalReferenceAssignedVariableName(node.parent);
+            if (modalVariableName) {
+                modalReferenceVariableNameToComponentClassNameMap.set(modalVariableName, componentSymbol);
+            }
+        };
+
+        /**
+         * Returns true if a class property is initialized via `input()` or `input.required()`.
+         */
+        const isSignalBackedProperty = member => {
+            const initializer = member.initializer;
+            if (!initializer || !ts.isCallExpression(initializer)) return false;
+            const expr = initializer.expression;
+            return (
+                (ts.isIdentifier(expr) && expr.text === "input") ||
+                (ts.isPropertyAccessExpression(expr) &&
+                    ts.isIdentifier(expr.expression) &&
+                    expr.expression.text === "input")
+            );
+        };
+
+
+        const findSignalBackedPropertyDeclaration = (componentSymbol, propertyName) => {
+            for (const decl of componentSymbol.getDeclarations() || []) {
+                if (!ts.isClassDeclaration(decl)) continue;
+                const member = decl.members.find(member =>
+                    ts.isPropertyDeclaration(member) &&
+                    ts.isIdentifier(member.name) &&
+                    member.name.text === propertyName &&
+                    isSignalBackedProperty(member)
+                );
+                if (member) return member;
+            }
+            return null;
+        };
+
+        /**
+         * Validate signal invocation (e.g. `this.courseId()`).
+         */
```

Comment:
We are checking here that `this.courseId()` is not how the signal is invoced, aren't we? I think the javadoc is misleading here, I would have misunderstood it